nasa-gibs/worldview
# WV-2126 Right Click Menu

## Description

Created a right click/context menu that gives the user some quick shortcuts to some commonly used features.

![image](https://user-images.githubusercontent.com/95651563/153051242-5434bfa2-3dfd-4e9d-9546-2537ae717068.png)

## How to Test
1. Run `git checkout wv-2126`
2. Run `npm ci && npm run build && npm start` to update your local environment and start up the local server.
3. Go to [localhost:3000](http://localhost:3000)
4. Right click anywhere on the map. Your OS/Browser context menu should not appear at all.  Instead, a semi-transparent dark gray box with white text and icons should appear, as shown in the screenshot above.
5. The following menu items should show and execute the following actions:

- Copy to Clipboard
  - Copies the coordinates that were collected during the right click directly to the user's clipboard
  - Flashes a tooltip above the right click menu confirming the coordinates have been copied
- Add a Place Marker
  - Adds a place marker at the coordinates that were collected during the right click
  - Appears at the top-left of the context menu
- Measure Distance in [unit]
  - Activates the distance measuring tool and goes into distance measuring "mode"
  - [unit] updates when "Change Units to..." is clicked in context menu
- Measure Area in [unit]
  - Activates the area measuring tool and goes into area measuring "mode"
  - [unit] updates when "Change Units to..." is clicked in context menu
- Remove Measurements
  - Clears all measurements on the map
- Change Units to [unit]
  - Toggles between "km" and "mi"
  - Little toggle icon is "off" when in "km"
  - Little toggle icon is "on" when in "mi"
